src/Compilers/CSharp/Portable/Emitter/Model/NamedTypeSymbolAdapter.cs

```diff
@@ -661,10 +661,13 @@ IEnumerable<Cci.IMethodDefinition> Cci.ITypeDefinition.GetMethods(EmitContext co
         {
             CheckDefinitionInvariant();
 
+            // All constructors in attributes should be emitted
+            bool isAttribute = DeclaringCompilation.IsAttributeType(this);
```

Comment:
> bool isAttribute = DeclaringCompilation.IsAttributeType(this); [](start = 11, length = 63)

We probably should avoid calculating this when nothing is going to be filtered out anyway.

Reply:
What's the best code pattern to do something like this? I feel like `Lazy` with a lambda is too heavyweight, and a `bool hasComputedIsAttribute; bool isAttribute;` is kind of gross (so I want confirmation before doing that). (Or did you mean something else by that? Not computing if `context.IncludePrivateMembers` is true or something like that?)

Reply:
> Or did you mean something else by that? Not computing if context.IncludePrivateMembers is true or something like that?)

Yes, something like that. Do not calculate if ```ShouldInclude``` doesn't filter anything out.

---
In reply to: [388878100](https://github.com/dotnet/roslyn/pull/42192#discussion_r388878100) [](ancestors = 388878100)

Reply:
I would also rename the local to something like ```alwaysIncludeConstructors``` it will cause less confusion once its value is conditionally calculated. The type doesn't stop to be an attribute no matter what ```ShouldInclude``` does.

---
In reply to: [389016262](https://github.com/dotnet/roslyn/pull/42192#discussion_r389016262) [](ancestors = 389016262,388878100)